cpp/src/arrow/compute/kernels/scalar_round.cc

```diff
@@ -906,9 +980,8 @@ void AddNullExec(ScalarFunction* func) {
   DCHECK_OK(func->AddKernel(std::move(input_types), OutputType(null()), NullToNullExec));
 }
 
-template <typename Op, typename FunctionImpl = ArithmeticFunction>
-std::shared_ptr<ScalarFunction> MakeArithmeticFunction(std::string name,
-                                                       FunctionDoc doc) {
+template <typename Op, typename FunctionImpl = RoundFunction>
+std::shared_ptr<ScalarFunction> MakeRoundFunction(std::string name, FunctionDoc doc) {
```

Comment:
You didn't add this but...is this used?  I think this file must have been generated from scalar_arithmetic.cc and not quite cleaned up all the way.

Reply:
Dropped.